### PR TITLE
8259550: The content of the print out displayed incomplete with the NimbusLAF

### PR DESCRIPTION
**Pull request by TejeshR13**
The issue is w.r.t to way that Synth prints the table. In Synth, the printing of Table happens column wise by painting blocks of columns which is due to the fact that Graphics used which is `ProxyGrpahics2D` (which get selected in `RasterPrinterJob `class, otherwise it would be `WPathGraphics` for other Look and feel).  Since the printing happens in block wise and the columns where the x position is greater than the `table.bounds` doesn't gets painted because of the intersection check condition `!bounds.intersects(clip)`. Hence only droplines are painted instead of cells and gridlines.  I couldn't find any reason for adding the intersection check here and when removed it didn't cause any regression too, rather it solves the issue and actually made sense too. In other Look and Feel its not required to apply the fix (in `BasicTableUI `class) because its not affecting anything yet since the painting happens for whole table. Hence proposing the fix for only SynthTableUI class.
([PrintManualTest_FitWidthMultiple.java](https://github.com/openjdk/jdk/blob/master/test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java)) test is been used for testing.

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8259550](https://bugs.openjdk.org/browse/JDK-8259550): The content of the print out displayed incomplete with the NimbusLAF (**Bug** - P4)

### Reviewers
 * [Damon Nguyen](https://openjdk.org/census#dnguyen) (@DamonGuy - Committer)
 * [Prasanta Sadhukhan](https://openjdk.org/census#psadhukhan) (@prsadhuk - **Reviewer**) ⚠️ Review applies to [bb22d322](https://git.openjdk.org/jdk/pull/17053/files/bb22d322c51eaef42c13e9d0220d882de9bc2a93)
 * [Abhishek Kumar](https://openjdk.org/census#abhiscxk) (@kumarabhi006 - Committer) ⚠️ Review applies to [628f657c](https://git.openjdk.org/jdk/pull/17053/files/628f657c55883808460a1f275113210046f8c211)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/17053/head:pull/17053` \
`$ git checkout pull/17053`

Update a local copy of the PR: \
`$ git checkout pull/17053` \
`$ git pull https://git.openjdk.org/jdk.git pull/17053/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 17053`

View PR using the GUI difftool: \
`$ git pr show -t 17053`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/17053.diff">https://git.openjdk.org/jdk/pull/17053.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/17053#issuecomment-1851305889)

### CODE REVIEW

**Comment by DamonGuy** [comment]
I see no difference in Nimbus LAF PrintManualTest_FitWidthMultiple.java with this change. Is there anything else that needs changing other than in SynthTableUI.java? Or any missing configuration on my end?

**Review by DamonGuy — COMMENTED** [review]
I see no difference in Nimbus LAF PrintManualTest_FitWidthMultiple.java with this change. Is there anything else that needs changing other than in SynthTableUI.java? Or any missing configuration on my end?

**Comment by TejeshR13** [comment]
> Is there anything else that needs changing other than in SynthTableUI.java? Or any missing configuration on my end?

Not any specific configuration is required, just can print it on a pdf for testing and see that few of the last column cells wont be printed without the fix.

**Comment by DamonGuy** [comment]
This was my output without the fix:
Screenshot 2024-01-16 at 3 32 33 PM

This was my output with the fix:
Screenshot 2024-01-16 at 3 33 13 PM

Comparing it to the description/example you gave on the difference, I can't seem to recreate that change. These screenshots were generated by using the ```Print``` option on the test and outputting to a PDF file. Any insights for why I'm not seeing the same thing?

**Comment by TejeshR13** [comment]
> This was my output without the fix: Screenshot 2024-01-16 at 3 32 33 PM
> 
> This was my output with the fix: Screenshot 2024-01-16 at 3 33 13 PM
> 
> Comparing it to the description/example you gave on the difference, I can't seem to recreate that change. These screenshots were generated by using the `Print` option on the test and outputting to a PDF file. Any insights for why I'm not seeing the same thing?



> This was my output without the fix: Screenshot 2024-01-16 at 3 32 33 PM
> 
> This was my output with the fix: Screenshot 2024-01-16 at 3 33 13 PM
> 
> Comparing it to the description/example you gave on the difference, I can't seem to recreate that change. These screenshots were generated by using the `Print` option on the test and outputting to a PDF file. Any insights for why I'm not seeing the same thing?

I guess the frame size when you give to print is quite more enough for it to print all of the columns. The size of frame where the issue can be reproduced is around 720x730 px. Can you check the frame size before giving it to print.

**Comment by prsadhuk** [comment]
I am also not able to reproduce the problem now with latest jdk without the fix and getting the same full output without any missing columns (same as damon is seeing).

**Review by prsadhuk — APPROVED** [review]
Please add this bugid to [PrintManualTest_FitWidthMultiple.java](https://github.com/openjdk/jdk/blob/master/test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java)

**Comment by prsadhuk** [comment]
I guess it will be better to add additional testing to this test in view of this fix so that it doesn't go under the radar in future incase something change in Nimbus or in printing code..
```
  * @run main/manual PrintManualTest_FitWidthMultiple
+ * @run main/othervm/manual -Dswing.defaultlaf=javax.swing.plaf.nimbus.NimbusLookAndFeel PrintManualTest_FitWidthMultipl
```

File: test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java [hunk 1]
```diff
@@ -22,7 +22,7 @@
  */
 /*
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java:23]: Can add a blank line before this.

**Comment by kumarabhi006** [comment]
Since you are updating the test file, you can check for the spacing as per coding standard.

**Comment by kumarabhi006** [comment]
> Comparing it to the description/example you gave on the difference, I can't seem to recreate that change. These screenshots were generated by using the `Print` option on the test and outputting to a PDF file. Any insights for why I'm not seeing the same thing?

I guess you are testing on Mac OS on which there is no difference in pdf files with or without fix. Able to reproduce on windows 11 and ubuntu 22.04 as mentioned.

**Comment by TejeshR13** [comment]
> > Comparing it to the description/example you gave on the difference, I can't seem to recreate that change. These screenshots were generated by using the `Print` option on the test and outputting to a PDF file. Any insights for why I'm not seeing the same thing?
> 
> I guess you are testing on Mac OS on which there is no difference in pdf files with or without fix. Able to reproduce on windows 11 and ubuntu 22.04 as mentioned.

@DamonGuy said he is not able to reproduce in Windows OS. Might be some settings related, not sure (As suggested in meeting might be of paper size A4).

**Comment by prsadhuk** [comment]
Please update the test with additional testing line as mentioned...

File: test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java [hunk 2]
```diff
@@ -20,12 +20,14 @@
...
+
 /*
  * @test
- * @bug 8170349
+ * @bug 8170349 8259550
  * @summary  Verify if printed content is within border and all columns are
  *           printed for PrintMode.FIT_WIDTH
- * @run main/manual PrintManualTest_FitWidthMultiple
+ * @run main/othervm/manual -Dswing.defaultlaf=javax.swing.plaf.nimbus.NimbusLookAndFeel PrintManualTest_FitWidthMultiple
+Merg
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java]: Looks like some typo mistake.

File: test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java [hunk 3]
```diff
@@ -114,38 +116,44 @@ public void run() {
...
-                JScrollPane scrollpane=new JScrollPane(table);
+                JScrollPane scrollpane = new JScrollPane(table);
                 fr = new JFrame("PrintManualTest_FitWidthMultiple");
                 fr.getContentPane().add(scrollpane);
 
                 /* Light Weight Panel for holding Print and other buttons */
-                JPanel btnpanel=new JPanel();
+                JPanel btnpanel = new JPanel();
                 btnpanel.add(printButton);
                 fr.getContentPane().add(btnpanel,BorderLayout.SOUTH);
```
Comment (kumarabhi006) [inline test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java:147]:
```suggestion
                fr.getContentPane().add(btnpanel, BorderLayout.SOUTH);
```
May be you can fix this here and other places as well. space after `,`.

**Review by kumarabhi006 — APPROVED** [review]
Minor comments else test is verified on windows and linux platform.

File: test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java [hunk 4]
```diff
@@ -20,12 +20,13 @@
...
  */
+
 /*
  * @test
- * @bug 8170349
+ * @bug 8170349 8259550
  * @summary  Verify if printed content is within border and all columns are
  *           printed for PrintMode.FIT_WIDTH
- * @run main/manual PrintManualTest_FitWidthMultiple
+ * @run main/othervm/manual -Dswing.defaultlaf=javax.swing.plaf.nimbus.NimbusLookAndFeel PrintManualTest_FitWidthMultiple
```
Comment (prsadhuk) [inline test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java:29]: You need to keep both not only Nimbus

Comment (TejeshR13) [reply test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java:29]: Meaning with default and Nimbus L&F?

Comment (prsadhuk) [reply test/jdk/javax/swing/JTable/PrintManualTest_FitWidthMultiple.java:29]: yes

**Review by DamonGuy — APPROVED** [review]
I re-ran doing the same as Prasanta on Windows 11. LGTM now.

**Pull request closed without merging**